src/onnxruntime_utils.h

```diff
@@ -37,13 +37,50 @@ namespace triton { namespace backend { namespace onnxruntime {
 
 extern const OrtApi* ort_api;
 
-#define RESPOND_ALL_AND_RETURN_IF_ERROR(RESPONSES, RESPONSES_COUNT, X) \
-  do {                                                                 \
-    TRITONSERVER_Error* raarie_err__ = (X);                            \
-    if (raarie_err__ != nullptr) {                                     \
-      SendErrorForResponses(RESPONSES, RESPONSES_COUNT, raarie_err__); \
-      return;                                                          \
-    }                                                                  \
+#define RESPOND_ALL_AND_RETURN_IF_ERROR(                            \
+    REQUESTS, REQUEST_COUNT, RESPONSES, S)                          \
+  do {                                                              \
+    TRITONSERVER_Error* raarie_err__ = (S);                         \
+    if (raarie_err__ != nullptr) {                                  \
+      SendErrorForResponses(RESPONSES, REQUEST_COUNT, raarie_err__);\
+      for (uint32_t r = 0; r < REQUEST_COUNT; ++r) {                \
+        if (RESPONSES[r] != nullptr) {                              \
```

Comment:
not sure how this works... 
SendErrorForResponses sets walks through each response and after response is sent sets it to nullptr

Check: https://github.com/triton-inference-server/tensorrt_backend/blob/main/src/tensorrt.cc#L93

Reply:
I believe it worked because dynamic batching was not being used. @jcwchen  Can you address @askhade's comment and verify your changes work even when dynamic batching is enabled?

Reply:
Trying. How can I enable dynamic batching? Follow the steps here? https://github.com/triton-inference-server/server/blob/main/docs/architecture.md Thanks

Reply:
So, yes.. Adding  `dynamic_batching { max_queue_delay_microseconds: 5000000}` in the model config will allow Triton to wait for 5sec for forming batches. 

Just make sure that you would need to send multiple inference requests to the server within the time interval so that they can be batched together. Best way would be to use perf_analyzer with --concurrency-range=<max_batch_size of the model>.

I would recommend keeping the max_batch_size of the model greater than or equal to 4.

Reply:
Thank you for the detailed instructions! Although I have tried other models (which did not crash before) with dynamic batching and every thing looks good, I cannot enable dynamic batching with my crashed model (it seems it does not support max_batch_size and always complains the shape is wrong)...  It's quite hard for me to verify my update with dynamic batching, but it can still resolve the crash issue without enabling dynamic batching for sure.

Thanks @askhade for pointing out and I think you are right so I have made send response error + request release be same as the reference code; also refactored the structure to shared the common code. Please review it again